src/AwsMetadata.js

```diff
@@ -53,6 +53,7 @@ export default class AwsMetadata {
       this.lookupMetadataKey(`network/interfaces/macs/${results.mac}/vpc-id`, (err, vpcId) => {
         results['vpc-id'] = vpcId;
         this.logger.debug('Found Instance AWS Metadata', results);
+        Object.keys(results).filter(k => results[k] === null).forEach(k => delete results[k]);
```

Comment:
The lint config will not accept == comparison. I have submitted an update explicitly also filtering out `undefined` and improved the test to check this.